data_steward/analytics/cdr_ops/cross_ct_rt/output_prod_publish_qc.py

```diff
@@ -449,3 +449,89 @@
+# - Example: Order by HPO site name and insert the first into the death table
+#
+# This QC confirms that the logic for the primary records are applied as expected in the `AOU_DEATH` table.
+
+# +
+query = JINJA_ENV.from_string("""
+WITH qc_aou_death AS (
+    SELECT 
+        aou_death_id, 
+        CASE WHEN aou_death_id IN (
+            SELECT aou_death_id FROM `{{project_id}}.{{dataset_id}}.aou_death`
+            QUALIFY RANK() OVER (
+                PARTITION BY person_id 
+                ORDER BY
+                    LOWER(src_id) NOT LIKE '%healthpro%' DESC, -- EHR records are chosen over HealthPro ones --
+                    death_date ASC, -- Earliest death_date records are chosen over later ones --
+                    death_datetime ASC NULLS LAST, -- Earliest non-NULL death_datetime records are chosen over later or NULL ones --
+                    src_id ASC -- EHR site that alphabetically comes first is chosen --
+            ) = 1   
+        ) THEN TRUE ELSE FALSE END AS primary_death_record
+    FROM `{{project}}.{{dataset}}.aou_death`    
+)
+SELECT ad.aou_death_id
+FROM `{{project_id}}.{{dataset}}.aou_death` ad
+LEFT JOIN qc_aou_death qad
+ON ad.aou_death_id = qad.aou_death_id
+WHERE ad.primary_death_record != qad.primary_death_record
+""").render(project_id=dest_project_id, dataset=dest_dataset_id)
+df = execute(client, query)
+
+success_msg = 'All death records have the correct `primary_death_record` values.'
+failure_msg = '''
+    <b>{code_count}</b> records do not have the correct `primary_death_record` values. 
+    Investigate and confirm if (a) any logic is incorrect, (b) the requirement has changed, or (c) something else.
+'''
+render_message(df,
+               success_msg,
+               failure_msg,
+               failure_msg_args={'code_count': len(df)})
+# -
+
+# # QC for DEATH table
+
+# From CDR V8, the DEATH table must exist and have the primary death records from AOU_DEATH in the output data stage.
+# This QC confirms that the DEATH table is there and has correct data.
+
+# +
+query_if_empty = JINJA_ENV.from_string("""
+WITH primary_aou_death AS (
+    SELECT person_id, death_date, death_datetime, death_type_concept_id, cause_concept_id, cause_source_value, cause_source_concept_id
+    FROM `{{project_id}}.{{dataset}}.aou_death`
+    WHERE primary_death_record = TRUE
+)
+(
+    SELECT person_id, death_date, death_datetime, death_type_concept_id, cause_concept_id, cause_source_value, cause_source_concept_id
+    FROM primary_aou_death
+    EXCEPT DISTINCT
+    SELECT person_id, death_date, death_datetime, death_type_concept_id, cause_concept_id, cause_source_value, cause_source_concept_id
+    FROM `{{project_id}}.{{dataset}}.death`
+)
+UNION ALL
+(
+    SELECT person_id, death_date, death_datetime, death_type_concept_id, cause_concept_id, cause_source_value, cause_source_concept_id
```

Comment:
Suggestion: You are combining two potential issues with the UNION.  A helper column might be nice in case of failure. One that defines which of the queries is returning the rows.  Something like:
"primary_records_missing_from_death" as issue
"unexpected_records_in_death" as issue